tarhelper/untar.go

```
@@ -258,11 +260,23 @@ func (u *Untar) processEntry(header *tar.Header) error {
 
 	// resolve the destination and then reset the name based on the resolution
 	destDir, err := u.resolveDestination(filepath.Dir(name))
-	name = filepath.Join(destDir, filepath.Base(name))
 	if err != nil {
 		return err
 	}
 
+	name = filepath.Join(destDir, filepath.Base(name))
+
+	// The path length of the extracted file might exceed Windows maximum of
+	// 260 chars.
+	if runtime.GOOS == "windows" {
+		absPath, err := filepath.Abs(name)
+		// If we can't get the absolute path, just assume it's less than 260
+		// chars. The assumption will be correct most of the time.
```

This is the relevant Windows code:
```go
func abs(path string) (string, error) {
	return syscall.FullPath(path)
}
```
So I guess it can fail because it's a syscall?

Here was my thinking for ignoring the error: A) If `filepath.Abs` returns an error and we return it, we fail with an inscrutable syscall error. B) If `filepath.Abs` returns and error and we ignore the error it AND `absPath` really is longer than 260 chars, we fail with an inscrutable "system cannot find the path specified" error. B is less likely because both conditions have to be true.

I could be wrong. I'm not sure if there's such a thing as a transient syscall error.